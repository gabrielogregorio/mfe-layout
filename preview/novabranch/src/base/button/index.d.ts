import { ReactNode } from 'react';
type ButtonProps = {
    onClick: () => void;
    children: ReactNode;
};
export declare const Button: ({ onClick, children }: ButtonProps) => import("react/jsx-runtime").JSX.Element;
export {};
