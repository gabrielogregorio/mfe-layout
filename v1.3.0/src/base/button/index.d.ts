import { ReactNode } from 'react';
type ButtonProps = {
    onClick: () => void;
    children: ReactNode;
};
export declare const Button: ({ onClick, children }: ButtonProps) => JSX.Element;
export {};
