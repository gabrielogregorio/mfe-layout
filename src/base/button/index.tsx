import { ReactNode } from 'react';

type ButtonProps = {
  onClick: () => void;
  children: ReactNode;
};

export const Button = ({ onClick, children }: ButtonProps) => (
  <button type="button" className="text-white text-[32px] uppercase w-full h-screen" onClick={() => onClick()}>
    {children}
  </button>
);
