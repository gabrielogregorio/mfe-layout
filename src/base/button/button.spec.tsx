import { fireEvent, render, screen } from '@testing-library/react';
import { Button } from '@/base/button';

describe('<Button />', () => {
  it('should render component with children', () => {
    render(<Button onClick={jest.fn}>Mock Content</Button>);
    expect(screen.getByRole('button', { name: 'Mock Content' })).toBeInTheDocument();
  });

  it('should be clicked', () => {
    const func = jest.fn();
    render(<Button onClick={func}>Conteudo2</Button>);

    const button = screen.getByRole('button');
    expect(func).toHaveBeenCalledTimes(0);

    fireEvent.click(button);
    fireEvent.click(button);

    expect(func).toHaveBeenCalledTimes(2);
  });

  it('should be snapshot', () => {
    const { container } = render(<Button onClick={jest.fn}>Conteudo2</Button>);

    expect(container).toMatchSnapshot();
  });
});
