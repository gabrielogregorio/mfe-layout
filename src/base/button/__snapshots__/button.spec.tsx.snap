// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should be snapshot 1`] = `
<div>
  <button
    class="text-white text-[32px] uppercase w-full h-screen"
    type="button"
  >
    Conteudo2
  </button>
</div>
`;
